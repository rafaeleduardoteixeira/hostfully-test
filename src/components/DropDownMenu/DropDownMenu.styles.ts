import styled from 'styled-components';
import { colors } from '../../themes';

export const Container = styled.div`
  position: absolute;
  width: 150px;
  right: 80px;
  top: 50px;
  background-color: white;
  border-radius: 10px;
  box-shadow: 0px 0px 10px 0px rgba(0, 0, 0, 0.1);
  padding: 10px;
  z-index: 10;
  color: ${colors.black};
  gap: 10px;
`;

export const Item = styled.div`
  padding: 10px;
  cursor: pointer;
  &:hover {
    border-radius: 10px;
    background-color: #f5f5f5;
  }
`;
