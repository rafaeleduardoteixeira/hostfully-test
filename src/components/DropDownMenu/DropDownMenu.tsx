import React from 'react';
import { useNavigate } from 'react-router-dom';
import { Container, Item } from './DropDownMenu.styles';

export const DropDownMenu = (): JSX.Element => {
  const navigate = useNavigate();

  const handleTrips = () => {
    navigate('/my-trips');
  };

  return (
    <Container>
      <Item onClick={handleTrips}>Trips</Item>
    </Container>
  );
};
