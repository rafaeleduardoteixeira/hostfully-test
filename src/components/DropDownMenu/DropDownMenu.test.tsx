import React from 'react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { DropDownMenu } from './DropDownMenu';

const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...(jest.requireActual('react-router-dom') as any),
  useNavigate: () => mockedUsedNavigate,
}));

describe('DropDownMenu', () => {
  test('Snapshot test', () => {
    const { asFragment } = render(<DropDownMenu />);
    expect(asFragment()).toMatchSnapshot();
  });
  test('Click on Trips menu option', () => {
    render(<DropDownMenu />);
    const linkElement = screen.getByText(/Trips/i);
    linkElement.click();
    expect(mockedUsedNavigate).toBeCalled();
  });
});
