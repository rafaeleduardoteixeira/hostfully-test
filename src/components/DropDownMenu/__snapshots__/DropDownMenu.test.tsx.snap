// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropDownMenu Snapshot test 1`] = `
<DocumentFragment>
  <div
    class="sc-braxZu bXpJGR"
  >
    <div
      class="sc-gJhJTp jzzMNQ"
    >
      Trips
    </div>
  </div>
</DocumentFragment>
`;
